### PR TITLE
Lobbyists suggester profile

### PR DESCRIPTION
**Pull request by laureenas**
The MVP that has been requested.

### CODE REVIEW

File: manoseimas/mps_v2/json_urls.py [hunk 1]
```diff
@@ -14,4 +14,6 @@
     url(r'^discussion/(?P<statement_id>\d+)/?$',
         views.mp_discussion_json, name='mp_discussion_json'),
     url(r'^suggesters/?$', views.suggesters_json, name='suggesters_json'),
+    url(r'^resolutions/(?P<suggester_slug>.+)/?$',
```
Comment (mgedmin) [inline manoseimas/mps_v2/json_urls.py]: This eats slashes too, suggest replacing `.+` with `[^/]`.

File: manoseimas/mps_v2/migrations/0031_suggester_slug_data.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+from django.db import models, migrations
+from django.template.defaultfilters import slugify
+
+
+def populate_slugs(apps, schema_editor):
+    # borrowed from http://stackoverflow.com/questions/13243625/generate-slug-field-data-for-existing-entries-in-database
+    Suggester = apps.get_model("mps_v2", "Suggester")
```
Comment (mgedmin) [inline manoseimas/mps_v2/migrations/0031_suggester_slug_data.py]: Ooh, nice trick!

File: manoseimas/mps_v2/static/components/filter_component.jsx [hunk 3]
```diff
@@ -117,7 +117,7 @@ var SortableList = React.createClass({
       var subtabs = this.props.sidebar_subtabs;
       showSidebar = (
         <SidebarSubtabs options={this.state.subtab_options}
-                        subtab_selected={this.state.subtab_selected}
+                        subtab_selected={subtabs.active_subtab}
```
Comment (mgedmin) [inline manoseimas/mps_v2/static/components/filter_component.jsx]: I'm not sure why these two changes are part of this pull request.  Mistake?

File: manoseimas/mps_v2/templates/suggester_profile.jade [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    script(src="{% static 'components/react_paginator.jsx' %}", type="text/jsx", charset="utf-8")
+    script(src="{% static 'components/react_loader.jsx' %}", type="text/jsx", charset="utf-8")
+    script(src="{% static 'components/law_projects_component.jsx' %}", type="text/jsx", charset="utf-8")
+
+- block header
+  - include "snippets/header_bootstrap.jade"
+  - include "header_scrooled.jade"
+
+- block middle
+  .photo-box.lobbyist
```
Comment (mgedmin) [inline manoseimas/mps_v2/templates/suggester_profile.jade]: The `lobbyist` class seems inappropriate here, maybe we can find a more generic term and rename it everywhere?

File: manoseimas/mps_v2/json_urls.py (hunk 1 above)
Comment (mgedmin) [inline manoseimas/mps_v2/json_urls.py]: Hm, I see older code has `url(r'^statements/(?P<mp_slug>.+)/$',`.

I hate that Django uses regular expressions for URL patterns.

I still prefer `(?P<...>[^/]+)/?$`, but `(?P<...>.+)/$` would also be acceptable.

I need to read about Django URL patterns and automated `/` appending and related topics.

File: manoseimas/mps_v2/views/json.py [hunk 5]
```diff
@@ -140,6 +146,26 @@ def suggesters_json(request):
                          'subtab_counts': subtab_counts()})
 
 
+def _mangle(text):
+    text = text.lstrip()
+    text = re.sub(u'^(PAGRINDINIO )?(KOMITETO )?(IŠVADA )?', u'', text)
+    text = text.lstrip()
```
Comment (mgedmin) [inline manoseimas/mps_v2/views/json.py]: You could merge the two `lstrips()` with the regexp:

`re.sub(ur'^\s*(PAGRINDINIO )?(KOMITETO )?(IŠVADA )?\s*', u'', text)`.

File: manoseimas/mps_v2/views/json.py [hunk 6]
```diff
@@ -140,6 +146,26 @@ def suggesters_json(request):
...
+def _resolution_dict(resolution):
+    return {
+        'title': _mangle(resolution.title),
+        'url': resolution.source,
+    }
+
+def resolutions_json(request, suggester_slug):
+    """Get resolutions for a given suggester."""
+    resolutions_qs = CommitteeResolution.objects.filter(
+        suggestion__submitter__slug=suggester_slug).distinct()
```
Comment (mgedmin) [inline manoseimas/mps_v2/views/json.py]: What does `.distinct()` do here?

File: manoseimas/mps_v2/views/suggesters.py [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+from manoseimas.mps_v2.models import Suggester
+
+
+def suggester_profile(request, suggester_slug):
+    """A profile view for a suggester."""
+
+    try:
+        suggester = Suggester.objects.get(slug=suggester_slug)
+    except Suggester.DoesNotExist:
+        # TODO: handle this case gracefully
```
Comment (mgedmin) [inline manoseimas/mps_v2/views/suggesters.py]: Use https://docs.djangoproject.com/en/1.8/topics/http/shortcuts/#get-object-or-404 ?

**Comment by mgedmin** [comment]
I don't understand some of the changes, but I trust you :)

Basically LGTM, with some minor changes needed.

I'd like it better if we were writing *tests* for view code.

File: manoseimas/mps_v2/static/components/filter_component.jsx (hunk 3 above)
Comment (laureenas) [inline manoseimas/mps_v2/static/components/filter_component.jsx]: Yeah one could switch to master, fix the bug there. No harm here.

File: manoseimas/mps_v2/templates/suggester_profile.jade (hunk 4 above)
Comment (laureenas) [inline manoseimas/mps_v2/templates/suggester_profile.jade]: Fell free.

File: manoseimas/mps_v2/views/json.py (hunk 5 above)
Comment (laureenas) [inline manoseimas/mps_v2/views/json.py]: Will the re be equivalent if there are more than one space between the words?

File: manoseimas/mps_v2/views/json.py (hunk 6 above)
Comment (laureenas) [inline manoseimas/mps_v2/views/json.py]: A good question.

Without `distinct` I would get multiple repetitions of resolution. I was wondering whether this has to do with the way django does lookups. Namely, perhaps it construct a natural join on the 3 tables and returns one row for each suggestion which means resolutions get duplicated.

Comment (mgedmin) [inline manoseimas/mps_v2/views/json.py]: When you say "multiple repetitions", do you mean multiple resolutions with the same title, or multiple resolutions with the same primary key?

Do we actually have duplicates in the database?

**Comment by laureenas** [comment]
I'l love to have view tests as well. A single integration test that covers 60% of the code would be way better than 0% coverage.

File: manoseimas/mps_v2/views/suggesters.py (hunk 7 above)
Comment (laureenas) [inline manoseimas/mps_v2/views/suggesters.py]: Thanks for digging this function up!

File: manoseimas/mps_v2/views/json.py (hunk 6 above)
Comment (laureenas) [inline manoseimas/mps_v2/views/json.py]: I got multiple resolutions with the same title and the same URL. I have not checked the key. 

Hey, let's look up the QS. Also, let'l look up the SQL.

Comment (laureenas) [inline manoseimas/mps_v2/views/json.py]:
```
>>> print CommitteeResolution.objects.filter(suggestion__submitter__slug=suggester_slug).query
SELECT 
    `mps_v2_committeeresolution`.`id`,
    `mps_v2_committeeresolution`.`source`,
    `mps_v2_committeeresolution`.`created_at`,
    `mps_v2_committeeresolution`.`modified_at`,
    `mps_v2_committeeresolution`.`source_id`,
    `mps_v2_committeeresolution`.`title`
    FROM 
        `mps_v2_committeeresolution
       ` INNER JOIN `mps_v2_suggestion`
             ON (`mps_v2_committeeresolution`.`source_id` = `mps_v2_suggestion`.`source_id` )
         INNER JOIN `mps_v2_suggestion_submitter`
             ON ( `mps_v2_suggestion`.`id` = `mps_v2_suggestion_submitter`.`suggestion_id` )
         INNER JOIN `mps_v2_suggester`
             ON ( `mps_v2_suggestion_submitter`.`suggester_id` = `mps_v2_suggester`.`id` )
    WHERE `mps_v2_suggester`.`slug` = a-mitasiunas
```
the `.distinct()

Comment (mgedmin) [inline manoseimas/mps_v2/views/json.py]: Ah, cool!

Comment (laureenas) [inline manoseimas/mps_v2/views/json.py]:
```
(Pdb) suggester_slug
u'energetikos-ministerija'
(Pdb) resolutions_qs.values('id', 'source_id', 'source')
[{'source_id': u'473263', 'source': u'http://www3.lrs.lt/pls/inter3/dokpaieska.showdoc_l?p_id=473263&p_tr2=2', 'id': 1505L}, {'source_id': u'473263', 'source': u'http://www3.lrs.lt/pls/inter3/dokpaieska.showdoc_l?p_id=473263&p_tr2=2', 'id': 1505L}, {'source_id': u'467020', 'source': u'http://www3.lrs.lt/pls/inter3/dokpaieska.showdoc_l?p_id=467020&p_tr2=2', 'id': 1324L}]
```

File: manoseimas/mps_v2/json_urls.py (hunk 1 above)
Comment (mgedmin) [inline manoseimas/mps_v2/json_urls.py]: I want this fixed before the release, or at least tested that it works correcly with both

    http://localhost:8000/.../statements/some-slug

and

    http://localhost:8000/.../statements/some-slug/

**Pull request merged**